bundles/net.tourbook/src/net/tourbook/ui/tourChart/SlideoutTourChartOptions.java

```diff
@@ -44,316 +44,350 @@
+      if (tcc == null) {
+         // this occurred when tour chart is empty
+         return;
+      }
 
-		tcc.isShowBreaktimeValues = isShowBreaktimeValues;
-		tcc.isSRTMDataVisible = isSrtmDataVisible;
-		tcc.xAxisTime = xAxisStartTime;
+      final boolean canShowTimeOnXAxis = tcc.isShowTimeOnXAxis;
+      final boolean canShowSRTMData = tcc.canShowSRTMData;
 
-		_chkShowBreaktimeValues.setSelection(isShowBreaktimeValues);
-		_chkShowSrtmData.setSelection(isSrtmDataVisible);
-		_chkShowStartTimeOnXAxis.setSelection(isTourStartTime);
-		_chkShowValuePointTooltip.setSelection(isShowValuePointTooltip);
+      _chkShowBreaktimeValues.setSelection(tcc.isShowBreaktimeValues);
 
-		// this is not set in saveState()
-		_prefStore.setValue(ITourbookPreferences.VALUE_POINT_TOOL_TIP_IS_VISIBLE, isShowValuePointTooltip);
+      _chkShowSrtmData.setEnabled(canShowSRTMData);
+      _chkShowSrtmData.setSelection(tcc.isSRTMDataVisible);
 
-		_gridUI.resetToDefaults();
+      _chkShowStartTimeOnXAxis.setEnabled(canShowTimeOnXAxis);
+      _chkShowStartTimeOnXAxis.setSelection(tcc.xAxisTime == X_AXIS_START_TIME.TOUR_START_TIME);
 
-		onChangeUI();
-	}
+      _chkShowValuePointTooltip.setSelection(_prefStore.getBoolean(//
+            ITourbookPreferences.VALUE_POINT_TOOL_TIP_IS_VISIBLE));
 
-	private void restoreState() {
+      _chkSelectAllTimeSlices.setSelection(_prefStore.getBoolean(//
+            ITourbookPreferences.TOGGLE_STATE_SELECT_INBETWEEN_TIME_SLICES));
 
-		final TourChartConfiguration tcc = _tourChart.getTourChartConfig();
+      _gridUI.restoreState();
+   }
 
-		if (tcc == null) {
-			// this occured when tour chart is empty
-			return;
-		}
+   private void saveState() {
 
-		final boolean canShowTimeOnXAxis = tcc.isShowTimeOnXAxis;
-		final boolean canShowSRTMData = tcc.canShowSRTMData;
+      final TourChartConfiguration tcc = _tourChart.getTourChartConfig();
 
-		_chkShowBreaktimeValues.setSelection(tcc.isShowBreaktimeValues);
+      final boolean isShowBreaktimeValues = _chkShowBreaktimeValues.getSelection();
+      final boolean isTourStartTime = _chkShowStartTimeOnXAxis.getSelection();
+      final boolean isSrtmDataVisible = _chkShowSrtmData.getSelection();
 
-		_chkShowSrtmData.setEnabled(canShowSRTMData);
-		_chkShowSrtmData.setSelection(tcc.isSRTMDataVisible);
+      final X_AXIS_START_TIME xAxisStartTime = isTourStartTime
+            ? X_AXIS_START_TIME.TOUR_START_TIME
+            : X_AXIS_START_TIME.START_WITH_0;
 
-		_chkShowStartTimeOnXAxis.setEnabled(canShowTimeOnXAxis);
-		_chkShowStartTimeOnXAxis.setSelection(tcc.xAxisTime == X_AXIS_START_TIME.TOUR_START_TIME);
+      /*
+       * Update pref store
+       */
+      _prefStore.setValue(ITourbookPreferences.GRAPH_IS_BREAKTIME_VALUES_VISIBLE, isShowBreaktimeValues);
+      _prefStore.setValue(ITourbookPreferences.GRAPH_IS_SRTM_VISIBLE, isSrtmDataVisible);
+      _prefStore.setValue(ITourbookPreferences.GRAPH_X_AXIS_STARTTIME, isTourStartTime);
```

Comment:
In the beginning of MT, I also used pref store fields but in this tour chart options dialog you can see the mixture of different state fields.

Reply:
Ok, good to know